schemas/service.yml

```diff
@@ -36,17 +36,31 @@
       description: >
         Name of the service data is collected from.
 
-        The name of the service is normally user given. This allows if two
-        instances of the same service are running on the same machine
-        they can be differentiated by the `service.name`.
+        The name of the service is normally user given. This allows for
+        distributed services that run on multiple hosts to correlate the
+        related instances based on the name.
 
-        Also it allows for distributed services that run on multiple hosts to
-        correlate the related instances based on the name.
-
-        In the case of Elasticsearch the service.name could contain the cluster
-        name. For Beats the service.name is by default a copy of the `service.type`
+        In the case of Elasticsearch the `service.name` could contain the cluster
+        name. For Beats the `service.name` is by default a copy of the `service.type`
         field if no name is specified.
 
+    - name: node.name
+      level: extended
+      type: keyword
+      example: "master"
+      short: Name of the service node.
+      description: >
+        Name of a service node.
+
+        This allows for two nodes of the same service running on the same
+        host to be differentiated. Therefore, `service.node.name` should
+        typically be unique across nodes of a given service.
+
+        In the case of Elasticsearch, the `service.node.name` could contain
+        the node name, e.g. "ingest" or "master".
```

Comment:
I would remove the "ingest" or "master" examples here. This will confuse people vs the Elasticsearch roles. This field is rather meant to hold something like "instance-0000000016", if someone were to use Elastic Cloud. But we don't need to have an explicit string example for the ES use case. I think establishing that an ES node name maps to `service.node.name` is sufficient.

The last paragraph still reads like APM documentation. Documentation of our products doesn't belong in ECS, it belongs in each product's documentation. However, as I've already mentioned, I like the thinking you were describing earlier. I think it makes sense for other general cases, such as parsing web access logs. The quotation under point 2 above was specifically meant to replace this paragraph. This description would actually map well to how APM behaves, and users will be able to follow this to implement compatible mappings in similar logs.

Reply:
> The quotation under point 2 above was specifically meant to replace the last paragraph. This description would actually map well to how APM behaves, and users will be able to follow this to implement compatible mappings in similar logs.

This quotation doesn't really captures what I mean with this addition:

> In cases where the service is not a clustered or doesn't have the concept of a node name

This is irrelevant- even if there is a concept of a node name, whatever sends the data for it (filebeat, metricbeat, APM agent) still needs to discover it and put it in a common place, where this field will come into play and be used for filtering/aggregations.

> the host name or container name can be used to distinguish running instances that make up this service.

This maps well to how APM behaves, but, even if true, it specifically describes places where we didn't need this field 🙂. We DO use the container name for containerized services and the host name for others. We needed something in addition for cases where two service nodes are not containerized and are colocated on the same host (eg for redundancy purposes).

I'll try again, without any APM specifics.